public/app/plugins/datasource/grafana-azure-monitor-datasource/resourcePicker/resourcePickerData.ts

```diff
@@ -364,6 +364,12 @@ export default class ResourcePickerData extends DataSourceWithBackend<AzureMonit
         supportedMetricNamespaces = supportedMetricNamespaces.concat(namespaceVals);
       }
     }
+
+    if (supportedMetricNamespaces.length === 0) {
+      throw new Error(
+        'No namespaces returned. Validate the datasource configuration is correct and required permissions have been granted.'
```

Comment:
I would give a bit more of info (for example, people may not now what a "namespace" is). Something like:

```
Unable to resolve a list of valid metric namespaces. Validate that the current data source have permissions to retrieve [metric namespaces?] for the desired subscriptions.
```

(not sure what's the right permissions that needs to be set)

Reply:
I think we need to be consistent with the documentation and suggest that the user has the `Reader` role on the subscriptions they're querying